google/datalab/bigquery/_view.py

```diff
@@ -186,7 +186,7 @@ def results(self, use_cache=True, dialect=None, billing_tier=None):
     """
     output_options = QueryOutput.table(use_cache=use_cache)
     return self._materialization.execute(output_options, dialect=dialect,
-                                               billing_tier=billing_tier).results
+                                               billing_tier=billing_tier).result()
```

Comment:
Otherwise, we'd be calling the base class' `result()` method, which isn't what we want to do here.
The `QueryJob` and `Job` classes need clear resolution in my opinion, they're too confusing side by side. I suggest changing `QueryJob.results` into `QueryJob.query_result()`, and `Job.result()` into `Job.job_result()` so that either can be called on the inherited type.
Another option would be to keep only one type `Job` that's aware of both async operations as well as GCP operations, and its `.results()` returns the GCP operation's results, or falls back to the async operation's.